repo_create/spec_creater.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python -tt
+
+import os
+import subprocess
+from rpmfluff import SimpleRpmBuild
+import ConfigParser
+
+dir_name = "test-1/spec"
+dir_rpm = "test-1/build"
+
+
+def ensure_dir(f):
+    if not os.path.exists(f):
+        os.makedirs(f)
+
+
+def rpm_init(function, args):
+    rpm_in = globals()[function](*args)
+    return rpm_in
+
+
+def rpm_set_parameter(rpm_in, opt, parameter):
+    getattr(rpm_in, opt)(parameter)
+    return rpm_in
+
+
+def spec_rpm_creator():
+    repo = ConfigParser.RawConfigParser()
+    repo.optionxform = str
+    repo.read("repo.ini")
+    for rpm_id in repo.sections():
+        options = repo.options(rpm_id)
+        if "SimpleRpmBuild" in options:
+            parameters = repo.get(rpm_id, "SimpleRpmBuild").split(',')
+            try:
+                parameters[3] = [parameters[3]]
+            except IndexError:
+                pass
+            name = parameters[0]
+            ver = parameters[1]
+            rel = parameters[2]
+            rpm = rpm_init("SimpleRpmBuild", parameters)
+            options.remove("SimpleRpmBuild")
+            if options:
+                for option in options:
+                    rpm = rpm_set_parameter(rpm, option, repo.get(rpm_id, option))
+        rpm.make()
+        ensure_dir(dir_name)
+        subprocess.check_call(['mv test-rpmbuild-' + name + '-' + ver + '-' + rel + '/' + name +
+                               '.spec ' + dir_name], shell=True)
+        ensure_dir(dir_rpm)
+        subprocess.check_call(['mv  test-rpmbuild-' + name + '-' + ver + '-' + rel +
+                               '/RPMS/noarch/* ' + dir_rpm], shell=True)
+        subprocess.check_call(['rm -r test-rpmbuild-' + name + '-' + ver + '-' + rel], shell=True)
+
+
+def create_repo():
+    subprocess.check_call(['createrepo_c -v ' + dir_rpm + '/ --no-database --simple-md-filenames'], shell=True)
+
+
+spec_rpm_creator()
+create_repo()
```

Comment:
I advise you to use rpmfluff to avoid doing your own parser but you still did that again :). Attach how the test definition look like, please.